### PR TITLE
Use java macros in java_tools.

### PR DESCRIPTION
**Pull request by iirina**
`@rules_java` repository is now available to all Bazel java users. There is the possibility to override the repository in the users' WORKSPACE file in order to safely migrate #8746.

Progress on #8746.

### CODE REVIEW

**Comment by iirina** [comment]
> Does this mean that java_tools will fetch rules_java?

Yes, after a java_tools release containing this PR the java_tools will fetch rules_java.

> Will rules_java be bundled in Bazel (even as a http_archive)?

Not necessarily. My plan was to migrate users to adding rules_java to their own WORKSPACE file. I'm trying to figure out the consequences of embedding `rules_java` into Bazel. 1) It's not clear to me if/how users can override `rules_java` repository. 2) It's not clear to me if there's a way to embed @rules_java without the archive being always fetched. I would like to avoid the "why-is-bazel-downloading-java-stuff-for-my-non-java-build" story.

> I'm wondering if the direction shouldn't be reversed, rules_java should fetch java_tools, wdyt?

In theory they both need each other. I don't see how we can avoid adding a repository dependency on ruels_java from java_tools. Until we migrate the rules to Starlark it doesn't make much sense for rules_java to embed/use java_tools.

**Comment by iirina** [comment]
@hlopko as discussed offline I've added `@rules_java` to the workspace suffix. PTAL.

File: src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE [hunk 1]
```diff
@@ -238,4 +238,11 @@ http_archive(
     ],
 )
 
+http_archive(
```
Comment (hlopko) [inline src/main/java/com/google/devtools/build/lib/bazel/rules/java/jdk.WORKSPACE]: If people want to specify their own rules_java, they will get a conflict here, right? Can you use maybe? example: https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/repository/local_config_platform.WORKSPACE;l=3?q=utils.bzl%5C%22,%20%5C%22maybe&ss=bazel

File: src/test/shell/bazel/bazel_java_tools_test.sh [hunk 2]
```diff
@@ -173,4 +173,20 @@ function test_java_tools_has_proguard() {
   expect_path_in_java_tools "java_tools/third_party/java/proguard/GPL.html"
 }
 
+function test_java_tools_toolchain_builds() {
```
Comment (hlopko) [inline src/test/shell/bazel/bazel_java_tools_test.sh]: Can you also add a test that rules_java can be overrided? e.g. https://source.bazel.build/bazel/+/master:src/test/shell/bazel/platforms_test.sh;l=51?q=platforms_test.sh

We could probably add a test for java_tools as well, but maybe in a separate PR.

File: tools/jdk/WORKSPACE.java_tools [hunk 3]
```diff
@@ -0,0 +1,8 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
```
Comment (hlopko) [inline tools/jdk/WORKSPACE.java_tools]: What's jdk.WORKSPACE for and what's WORKSPACE.java_tools for? Do we need both of them to define rules_java?

File: src/test/shell/bazel/bazel_java_tools_test.sh (hunk 2 above)
Comment (iirina) [reply src/test/shell/bazel/bazel_java_tools_test.sh]: Sure. I added a new file src/test/shell/bazel/bazel_rules_java_test.sh.

**Comment by iirina** [comment]
@hlopko you can take another look.

**Pull request closed without merging**